### PR TITLE
✨ v3: Change startup message

### LINKED ISSUE #2032: 🚀 v3 Request: Revamp Startup Message
### Feature Description

I think we should revamp Fiber's startup message in v3 with more beautiful one. If you have some ideas please share them :smile:  :smile:  :smile: 

### Additional Context (optional)

_No response_

### Code Snippet (optional)

```go
package main

import "github.com/gofiber/fiber/v2"
import "log"

func main() {
  app := fiber.New()

  // An example to describe the feature

  log.Fatal(app.Listen(":3000"))
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my suggestion prior to opening this one.
- [X] I understand that improperly formatted feature requests may be closed without explanation.

### PR DESCRIPTION
**Pull request by gozeloglu**
Signed-off-by: Gökhan Özeloğlu <gokhan.ozeloglu@deliveryhero.com>

## Description

I changed the startup message. Figlet text is added. 

Fixes #2032

### CODE REVIEW

**Comment by efectn** [comment]
What do you think about the location of information messages?

**Comment by gozeloglu** [comment]
> What do you think about the location of information messages?

I think they can be under the `Fiber` text.

**Comment by efectn** [comment]
> > What do you think about the location of information messages?
> 
> I think they can be under the `Fiber` text.

Maybe they can be written like `=> server started on http://127.0.0.1:3000` like logging. What do you mean?

**Comment by efectn** [comment]
> > > What do you think about the location of information messages?
> > 
> > 
> > I think they can be under the `Fiber` text.
> 
> Maybe they can be written like `=> server started on http://127.0.0.1:3000` or  https://raw.githubusercontent.com/rs/zerolog/master/pretty.png as similar logging messages. What do you mean?

**Comment by gozeloglu** [comment]
> I think they can be under the `Fiber` text.
> 
> Maybe they can be written like `=> server started on http://127.0.0.1:3000` or https://raw.githubusercontent.com/rs/zerolog/master/pretty.png as similar logging messages. What do you mean?

Both options are cool. I am going to try to implement the second option, logging one.

**Comment by gozeloglu** [comment]
I didn't add color. I can use [fatih/color](https://github.com/fatih/color) or any other package. What do you think, @efectn?

**Comment by efectn** [comment]
> I didn't add color. I can use [fatih/color](https://github.com/fatih/color) or any other package. What do you think, @efectn?

You can use colors like old style. I think we don't need extra dep

**Comment by gozeloglu** [comment]
> > I didn't add color. I can use [fatih/color](https://github.com/fatih/color) or any other package. What do you think, @efectn?
> 
> You can use colors like old style. I think we don't need extra dep

I added colors to log messages. You can check an example output below. What do you think?

image

File: listen.go [hunk 1]
```diff
@@ -226,26 +242,36 @@ func (app *App) startupMessage(addr string, tls bool, pids string) {
 		procs = "1"
 	}
 
-	out := colorable.NewColorableStdout()
+	out := os.Stdout
 	if os.Getenv("TERM") == "dumb" || os.Getenv("NO_COLOR") == "1" || (!isatty.IsTerminal(os.Stdout.Fd()) && !isatty.IsCygwinTerminal(os.Stdout.Fd())) {
-		out = colorable.NewNonColorable(os.Stdout)
+		//out = colorable.NewNonColorable(os.Stdout)
```
Comment (efectn) [inline listen.go:247]: why

File: listen.go [hunk 2]
```diff
@@ -30,6 +30,22 @@ const figletFiberText = `
...
 /_/   /_/_.___/\___/_/     `
 
+var (
+	resetColor = "\033[0m"
+	red        = "\033[31m"
+	green      = "\033[32m"
+	blue       = "\033[34m"
+)
+
+func init() {
```
Comment (efectn) [inline listen.go:40]: get these colors from config

**Review by efectn — COMMENTED** [review]
maybe we can write version right of the `Fiber ASCII art`

**Review by efectn — COMMENTED** [review]
I think `server started on` message can be first or last message. Also what about adding a separator between ASCII and info messages like Echo v4.

File: listen.go (hunk 1 above)
Comment (gozeloglu) [reply listen.go:247]: When I used `out = colorable.NewNonColorable(os.Stdout)`, the colors weren't working. The output wasn't changing. Then. I tried this one and it worked  ¯\_(ツ)_/¯

File: listen.go (hunk 2 above)
Comment (gozeloglu) [reply listen.go:40]: OK, I'll take a look tomorrow.

**Comment by gozeloglu** [comment]
> maybe we can write version right of the `Fiber ASCII art`

```
    _______ __                      _____  ____   ____        __         __       
   / ____(_) /_  ___  _____   _   _|__  / / __ \ / __ \      / /_  ___  / /_____ _
  / /_  / / __ \/ _ \/ ___/  | | / //_ < / / / // / / /_____/ __ \/ _ \/ __/ __ `/
 / __/ / / /_/ /  __/ /      | |/ /__/ // /_/ // /_/ /_____/ /_/ /  __/ /_/ /_/ / 
/_/   /_/_.___/\___/_/       |___/____(_)____(_)____/     /_.___/\___/\__/\__,_/  
                                                                                  
```

It is like this. It can be confusing, but not a big deal.

**Comment by gozeloglu** [comment]
> I think `server started on` message can be first or last message. Also what about adding a separator between ASCII and info messages like Echo v4.

What do you mean by "separator between ASCII and info messages"? I don't know the Echo version. I just saw the example [here](https://echo.labstack.com/). Do you mean `----------------` between fancy text and logs?

File: listen.go (hunk 1 above)
Comment (gozeloglu) [reply listen.go:247]: Oh, I think you mean why I just commented out the line. Maybe I forgot to restore it.

**Comment by efectn** [comment]
> > maybe we can write version right of the `Fiber ASCII art`
> 
> ```
>     _______ __                      _____  ____   ____        __         __       
>    / ____(_) /_  ___  _____   _   _|__  / / __ \ / __ \      / /_  ___  / /_____ _
>   / /_  / / __ \/ _ \/ ___/  | | / //_ < / / / // / / /_____/ __ \/ _ \/ __/ __ `/
>  / __/ / / /_/ /  __/ /      | |/ /__/ // /_/ // /_/ /_____/ /_/ /  __/ /_/ /_/ / 
> /_/   /_/_.___/\___/_/       |___/____(_)____(_)____/     /_.___/\___/\__/\__,_/  
>                                                                                   
> ```
> 
> It is like this. It can be confusing, but not a big deal.

No. As a text, it's very big if we do it ASCII art.

> > I think `server started on` message can be first or last message. Also what about adding a separator between ASCII and info messages like Echo v4.
> 
> What do you mean by "separator between ASCII and info messages"? I don't know the Echo version. I just saw the example [here](https://echo.labstack.com/). Do you mean `----------------` between fancy text and logs?

Yes

File: listen.go (hunk 1 above)
Comment (gozeloglu) [reply listen.go]: I did something on my last commit, but I didn't get this if-block's logic.

**Comment by gozeloglu** [comment]
This is the last output of the startup message. 

image

**Comment by efectn** [comment]
> This is the last output of the startup message.
> 
> image

Will check it next 2-3 days. Looks nice to me

**Comment by efectn** [comment]
I made a little improvements and reverted go-colorable part. It was worked on me without any problem. Maybe it can be related about your terminal.

**Review by efectn — APPROVED** [review]

**Comment by ReneWerner87** [comment]
@jozsefsallai what do you think about the new start message, from the code it looks good, just don't know if the design is technically good, so i'm asking someone who knows something about the subject

**Review by ReneWerner87 — APPROVED** [review]

File: listen_test.go [hunk 3]
```diff
@@ -372,17 +373,49 @@ func Test_Listen_Master_Process_Show_Startup_MessageWithAppName(t *testing.T) {
...
+		EnablePrefork: true,
+	}
+
+	appName := "Serveur de vérification des données"
+	app := New(Config{AppName: appName})
+
+	startupMessage := captureOutput(func() {
+		app.startupMessage(":3000", false, "", cfg)
+	})
+	fmt.Println(startupMessage)
```
Comment (trunov) [inline listen_test.go:388]: why would you keep fmt.Println ?

Comment (efectn) [reply listen_test.go:388]: > why would you keep fmt.Println ?

We plan to remove most of them later to make testing messages cleaner.

**Comment by gozeloglu** [comment]
Hi @efectn and @ReneWerner87. When do you plan to merge this PR?

**Comment by ReneWerner87** [comment]
now
just wanted to wait for the feedback
but we can also change this shortly before version3, if there are any comments

**Pull request merged**